libbeat/management/management.go

```diff
@@ -58,7 +58,30 @@ func Register(name string, fn FactoryFunc, stability feature.Stability) {
 
 // Factory retrieves config manager constructor. If no one is registered
 // it will create a nil manager
-func Factory() FactoryFunc {
+func Factory(cfg *common.Config) FactoryFunc {
+	type modeConfig struct {
+		Mode string `config:"mode" yaml:"mode"`
+	}
+
+	modeCfg := &modeConfig{}
+	if err := cfg.Unpack(&modeCfg); err != nil {
+		return nilFactory
+	}
+
+	if modeCfg.Mode == "fleet" {
+		fleetFeat, err := feature.GlobalRegistry().Lookup(Namespace, "x-pack-fleet")
+		if err != nil {
+			return nilFactory
+		}
+
+		factory, ok := fleetFeat.Factory().(FactoryFunc)
+		if !ok {
+			return nilFactory
+		}
+
+		return factory
+	}
+
```

Comment:
We should make it pluggable so the Elastic License code doesn't impact the APL code.

Reply:
@michalpristas Just to clarify, I believe it would be better if we do not have any mention of fleet in the APL code I was referring to `fleetFeat, err := feature.GlobalRegistry().Lookup(Namespace, "x-pack-fleet")` See the `x-pack-fleet` section, If we make the lookup using a dynamic string we can remove that.

Reply:
👍  thanks for clarification

Reply:
the problem here is that if mode is not configured random factory will be picked instead of CM which is intended default.
we discussed different ways with @ph about how to handle default plugin which does not need to be in OSS and agreed on modifying factory in a way that it will receive configuration and return nil if it cannot respond to it. 
here we will try to instantiate factory up to the point where we receive not nil factory function. first responder still wins